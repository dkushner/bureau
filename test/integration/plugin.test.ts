import { createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Bureau from '@/index'

describe('installing Vue plugin', () => {
  const localVue = createLocalVue()
  localVue.use(Vuex)
  localVue.use(Bureau, { store: new Vuex.Store({ }) })

  const components = Object.keys(localVue.options.components).sort()

  it('should register all components', () => {
    expect(components).toMatchSnapshot()
  })

  it('should register all components in PascalCase', () => {
    components.forEach(name => {
      expect(name).toMatch(/^(?:[A-Z][a-z]*)+$/)
    })
  })
})