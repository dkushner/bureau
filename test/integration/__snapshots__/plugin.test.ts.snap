// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`installing Vue plugin should register all components 1`] = `
Array [
  "BureauForm",
]
`;
